core/corehttp/routing.go

```diff
@@ -0,0 +1,98 @@
+package corehttp
+
+import (
+	"context"
+	"net"
+	"net/http"
+	"time"
+
+	"github.com/ipfs/boxo/routing/http/server"
+	"github.com/ipfs/boxo/routing/http/types"
+	"github.com/ipfs/boxo/routing/http/types/iter"
+	cid "github.com/ipfs/go-cid"
+	core "github.com/ipfs/kubo/core"
+	"github.com/libp2p/go-libp2p/core/peer"
+	"github.com/libp2p/go-libp2p/core/routing"
+	"github.com/multiformats/go-multiaddr"
+)
+
+func RoutingOption() ServeOption {
+	return func(n *core.IpfsNode, _ net.Listener, mux *http.ServeMux) (*http.ServeMux, error) {
+		handler := server.Handler(&contentRouter{n})
+		mux.Handle("/routing/v1/", handler)
+		return mux, nil
+	}
+}
+
+type contentRouter struct {
+	n *core.IpfsNode
+}
+
+func (r *contentRouter) FindProviders(ctx context.Context, key cid.Cid) (iter.ResultIter[types.ProviderResponse], error) {
+	ctx, cancel := context.WithCancel(ctx)
+	ch := r.n.Routing.FindProvidersAsync(ctx, key, 20) // TODO: magic number, where to get this from
+	return iter.ToResultIter[types.ProviderResponse](&peerChanIter{
+		ch:     ch,
+		cancel: cancel,
+	}), nil
+}
+
+func (r *contentRouter) ProvideBitswap(ctx context.Context, req *server.BitswapWriteProvideRequest) (time.Duration, error) {
+	return 0, routing.ErrNotSupported
+}
+
+func (r *contentRouter) Provide(ctx context.Context, req *server.WriteProvideRequest) (types.ProviderResponse, error) {
+	return nil, routing.ErrNotSupported
+}
+
+type peerChanIter struct {
+	ch     <-chan peer.AddrInfo
+	cancel context.CancelFunc
+	next   *peer.AddrInfo
+}
+
+func (it *peerChanIter) Next() bool {
+	addr, ok := <-it.ch
+	if ok {
+		it.next = &addr
+		return true
+	} else {
+		it.next = nil
+		return false
+	}
+}
+
+func (it *peerChanIter) Val() types.ProviderResponse {
+	if it.next == nil {
+		return nil
+	}
+
+	rec := &providerRecord{
+		Protocol: "transport-unknown",
+		Schema:   "unknown",
```

Comment:
We don't know the protocol (or schema) of these peers. it is not specified how to indicate this.

Reply:
The current [specification](https://specs.ipfs.tech/routing/http-routing-v1/#known-transfer-protocols) does not really specify what to do for unknown protocols. Besides, in Kubo, when asking for providers, we also don't know for sure what protocol it is. It is possibly Bitswap, but we don't have a way of guaranteeing it. How to handle this is not yet known.

cc @lidel @masih